
---
title: "BubbleTree Tutorial"
author: "Wei Zhu, Michael Kuziora, Todd Creasy, Brandon Higgs"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: > 
    %\VignettePackage{BubbleTree}
    %\VignetteEngine{knitr::rmarkdown}
    %\VignetteIndexEntry{BubbleTree Tutorial}
    %\VignetteKeyword{CNV}
    %\VignetteKeyword{CopyNumberVariation}
    \usepackage[utf8]{inputenc}
---

```{r style, echo = FALSE, results = 'asis'}
BiocStyle::markdown()
```

## Introduction

To evaluate the aneuploidy and prevalence of clonal or quasiclonal tumors, we developed a novel tool to characterize the mosaic tumor genome on the basis of one major assumption: the genome of a heterogeneous multi-cell tumor biopsy can be sliced into a chain of segments that are characterized by homogeneous somatic copy number alternations (SCNAs) and B allele frequencies (BAFs). The model, termed BubbleTree, utilizes both SCNAs and the interconnected BAFs as markers of tumor clones to extract tumor clonality estimates. BubbleTree is an intuitive and powerful approach to jointly identify ASCN, tumor purity and (sub)clonality, which aims to improve upon current methods to characterize the tumor karyotypes and ultimately better inform cancer diagnosis, prognosis and treatment decisions.

## Quickstart to Using BubbleTree

To perform a BubbleTree analysis, data pertaining to the position and B allele frequency of heterozygous snps in the tumor sample and segmented copy number information including the position, number of markers/segment and log2 copy number ratio between tumor and normal samples must first be obtained. 

## Preparing Data for BubbleTree

BubbleTree was developed using both whole exome sequencing (WES) and whole genome sequening (WGS) NGS data from paired tumor/normal biopsies, but this model should also be applicable to array comparative genomic hybridization (aCGH) and single nucleotide polymorphism (SNP) array data.

There are many methods for generating and processing sequencing data in preparation for BubbleTree analysis. In the following section we provide example workflows starting from WES NGS which can be adapted as needed to alternate inputs.

### Preparing Sequence Variation Data

The primary BubbleTree requirement for sequence variant information is a `r Biocpkg("GRanges")` object containing the B alelle frequencies and genomic positions of variants known to be heterozygous in the paired normal sample.

Mapped reads from tumor and normal tissue can be processed with mutation caller software such as VARSCAN or MUTECT. In this example, we use a hypothetical vcf file output which contains mutation calls from both normal and tumor samples.

The B-allele frequency data is extracted using the Bioconductor package `r Biocpkg("VariantAnnotation")` and converted from string to numeric format.

Example data in the desired format is provided as part of this package as GRanges objects and can be loaded as follows.

```{r}
library(BubbleTree)
```

allCall.lst is pre-calculated CNV data from...
allRBD.lst is pre-calculated CNV data but ...

```{r}
load(system.file("data", "allCall.lst.RData", package="BubbleTree"))
load(system.file("data", "allRBD.lst.RData", package="BubbleTree"))
```

The remaining datasets used to support the CNV data display on the BubbleTree plots.

```{r}
load(system.file("data", "cancer.genes.minus2.rda", package="BubbleTree")) # list of 379 known cancer genes
load(system.file("data", "vol.genes.rda", package="BubbleTree")) # another list of 105 known cancer genes
load(system.file("data", "gene.uni.clean.gr.rda", package="BubbleTree"))
load(system.file("data", "cyto.gr.rda", package="BubbleTree")) # load cytoband coordinate data
load(system.file("data", "centromere.dat.rda", package="BubbleTree")) # load centromere coordinate data
load(system.file("data", "all.somatic.lst.RData", package="BubbleTree")) # load SNV location data
load(system.file("data", "allHetero.lst.RData", package="BubbleTree")) # load sequence variants
load(system.file("data", "allCNV.lst.RData", package="BubbleTree")) # load copy number variation data
load(system.file("data", "hg19.seqinfo.rda", package="BubbleTree")) # load hg19 sequence data
```

This will create BubbleTree plots for all 41 samples using adjusted and non-adjusted CNV data.

```{r}
library(BubbleTree)
load(system.file("data", "allCall.lst.RData", package="BubbleTree"))

btreeplotter <- new("BTreePlotter", max.ploidy=5, max.size=10)

pdf.filename <- paste("adjplot_all", "pdf", sep=".")
pdf(pdf.filename, width=6, height=4)

l_ply(names(allCall.lst), function(nn){
    cat("Processing sample name: ", nn, "\n")
    rbd1 <- allCall.lst[[nn]]@rbd
    rbd2 <- allCall.lst[[nn]]@rbd.adj
    arrows <- trackBTree(btreeplotter, rbd1, rbd2, min.srcSize=0.01, min.trtSize=0.01)
    btree <- drawBTree(btreeplotter, rbd1) + drawBubbles(btreeplotter, rbd2, "gray80") + arrows
    print(btree)
})
dev.off()

```

This will create BubbleTree plots for all 41 samples using only non-adjusted CNV data.

```{r}
library(BubbleTree)
load(system.file("data", "allCall.lst.RData", package="BubbleTree"))

btreeplotter <- new("BTreePlotter", max.ploidy=5, max.size=10)

pdf.filename <- paste("adjplot_single", "pdf", sep=".")
pdf(pdf.filename, width=8, height=6)

l_ply(names(allCall.lst), function(nn){
    cat("Processing sample name: ", nn, "\n")
    rbd1 <- allCall.lst[[nn]]@rbd
    rbd2 <- allCall.lst[[nn]]@rbd.adj
    arrows <- trackBTree(btreeplotter, rbd1, rbd2, min.srcSize=0.1, min.trtSize=0.1)
    btree <- drawBTree(btreeplotter, rbd1)  + arrows
    print(btree)
})
dev.off()

```

To generate a Excel report of a BubbleTree analysis. This report only shows samples that have tumors with high ploidy and high purity. 

```{r}
library(BubbleTree)
load(system.file("data", "allRBD.lst.RData", package="BubbleTree"))
btreepredictor <- new("BTreePredictor")
btreepredictor@config$cutree.h <- 0.15

high.ploidy <- rep(TRUE, length(allRBD.lst))
high.purity <- rep(TRUE, length(allRBD.lst))

high.ploidy[c("sam6",
              "ovary.wgs",
              "ovary.wes",
              "TCGA-06-0145-01A-01W-0224-08",
              "TCGA-13-1500-01A-01D-0472-01",
              "TCGA-AO-A0JJ-01A-11W-A071-09")] <- FALSE

high.purity[c("sam6", "ovary.wgs", "ovary.wes")] <- FALSE


allCall.lst <- plyr::llply(names(allRBD.lst), function(nn) {
    cat("Processing sample name: ", nn, "\n")
    rbd <- allRBD.lst[[nn]]
    btreepredictor@config$high.ploidy <- high.ploidy[nn]
    btreepredictor@config$high.purity <- high.purity[nn]
    btreepredictor <- loadRBD(btreepredictor, rbd)
    btreepredictor@config$min.segSize <- ifelse(max(btreepredictor@rbd$seg.size, na.rm=TRUE) < 0.4, 0.1, 0.4)
    
    btreepredictor <- btpredict(btreepredictor)
    
    cat(info(btreepredictor), "\n")
    return(btreepredictor)
})

names(allCall.lst) <- names(allRBD.lst)
results <- list()
for (name in names(allCall.lst)) {
    results[[name]] <- allCall.lst[[name]]@result$dist
}

xls.filename <- paste("all_calls_report", "xlsx", sep=".")
print(xls.filename)
saveXLS(results, xls.filename)

```

To perform a BubbleTree analysis with an overlay of 77 common cancer genes.

```{r}
library(BubbleTree)
load(system.file("data", "allCall.lst.RData", package="BubbleTree"))
load(system.file("data", "cancer.genes.minus2.rda", package="BubbleTree"))
load(system.file("data", "vol.genes.rda", package="BubbleTree"))
load(system.file("data", "gene.uni.clean.gr.rda", package="BubbleTree"))
load(system.file("data", "cyto.gr.rda", package="BubbleTree"))

# 77 common cancer genes
comm <- btcompare(vol.genes, cancer.genes.minus2)

btreeplotter <- new("BTreePlotter", branch.col="gray50")
annotator <-new("Annotate")

pdf.filename <- paste("bubbletree_preview", "pdf", sep=".")
pdf(pdf.filename, width=10, height=6)

allPreview <- llply(names(allCall.lst), function(nn){
    cat("Processing sample name: ", nn, "\n")
    cc <- allCall.lst[[nn]]
    z <- drawBTree(btreeplotter, cc@rbd.adj) + ggplot2::labs(title=sprintf("%s (%s)", nn, info(cc)))
    print(z)
    out <- cc@result$dist  %>% filter(seg.size >= 0.1 ) %>% arrange(gtools::mixedorder(as.character(seqnames)), start)
    
    ann <- with(out, {
        annoByGenesAndCyto(annotator,
                           as.character(out$seqnames),
                           as.numeric(out$start),
                           as.numeric(out$end),
                           comm$comm,
                           gene.uni.clean.gr=gene.uni.clean.gr,
                           cyto.gr=cyto.gr)
    })
    
    out$cyto <- ann$cyto
    out$genes <- ann$ann
    return(out)
})
dev.off()

```

BubbleTree can create a summary visualization that displays the concordance between copy number and max B-allele Frequency for each chromosome as well as comopare the BAF and R scores.

```{r}
library(BubbleTree)
load(system.file("data", "allCall.lst.RData", package="BubbleTree"))
load(system.file("data", "centromere.dat.rda", package="BubbleTree"))
load(system.file("data", "all.somatic.lst.RData", package="BubbleTree"))
load(system.file("data", "allHetero.lst.RData", package="BubbleTree"))
load(system.file("data", "allCNV.lst.RData", package="BubbleTree"))
load(system.file("data", "hg19.seqinfo.rda", package="BubbleTree"))

trackplotter <- new("TrackPlotter")

gr2 = centromere.dat

pdf.filename <- paste("all_tracks3", "pdf", sep=".")
pdf(pdf.filename, width=10, height=7)

plyr::l_ply(names(allCall.lst), function(nn) {
    cat("Processing sample name: ", nn, "\n")
    ymax <- ifelse(nn %in% c("lung.wgs", "lung.wes"), 9, 4.3)
    
    p1 <- xyTrack(trackplotter,
                  result.dat=allCall.lst[[nn]]@result,
                  gr2=gr2,
                  ymax=ymax) + ggplot2::labs(title=nn)
    
    p2 <- bafTrack(trackplotter,
                   result.dat=allCall.lst[[nn]]@result,
                   gr2=gr2,
                   somatic.gr=all.somatic.lst[[nn]])
    
    t1 <- getTracks(p1, p2)
    
    z1 <- heteroLociTrack(trackplotter, allCall.lst[[nn]]@result, gr2, allHetero.lst[[nn]])
    z2 <- RscoreTrack(trackplotter, allCall.lst[[nn]]@result, gr2, allCNV.lst[[nn]])
    t2 <- getTracks(z1, z2)
    
    gridExtra::grid.arrange(t1,t2, ncol=1)
    
})
dev.off()

```

To perform a comparison of cancer datasets.

```{r}
library(BubbleTree)
load(system.file("data", "allCall.lst.RData", package="BubbleTree"))

btp <- new("BTreePlotter", max.ploidy=5, max.size=10)
pairs <- list(c("lung.wes", "lung.wgs"),
              c("HCC4.Primary.Tumor", "HCC4.Recurrent.Tumor"),
              c("HCC11.Primary.Tumor", "HCC11.Recurrent.Tumor" ))

pdf("medi.comparePlot.single2.pdf", width=8,height=6)

plyr::l_ply(pairs, function(p){
    cat(p, "\n")
    rbd1 <- allCall.lst[[p[1]]]@result$dist
    rbd2 <- allCall.lst[[p[2]]]@result$dist
    
    srcSize <- 0.5
    trtSize <- ifelse(p[1] == "lung.wes", 0.1, 1)
    minOver <- ifelse(p[1] == "lung.wes", 5e6, 1e7)
    arrows <- trackBTree(btp,
                         rbd1,
                         rbd2,
                         min.srcSize=srcSize,
                         min.trtSize=trtSize,
                         min.overlap=minOver)
    
    z <- drawBTree(btp, rbd1)
    if(!is.null(arrows))
        z <- z + arrows + ggplot2::labs(title=sprintf("%s -> %s", p[1], p[2]))
    print(z)
})
dev.off()

```

To print out an Excel document of summary of the pre-called CNV data.

```{r}
library(BubbleTree)
load(system.file("data", "allCall.lst.RData", package="BubbleTree"))

all.summary <- plyr::ldply(allCall.lst, function(.Object) {
    purity <- .Object@result$prev[1]
    adj <- .Object@result$ploidy.adj["adj"]
    ploidy <- (2*adj -2)/purity + 2  # when purity is low the calculation result is not reliable
    
    with(.Object@result,
         return(c(Purity=round(purity,3),
                  Prevalences=paste(round(prev,3), collapse=", "),
                  "Tumor ploidy"=round(ploidy,1))))
}) %>% plyr::rename(c(".id"="Sample"))

xls.filename <- paste("all_summary", "xlsx", sep=".")
saveXLS(list(Summary=all.summary), xls.filename)

```
