<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Wei Zhu, Michael Kuziora, Todd Creasy, Brandon Higgs" />

<meta name="date" content="2015-08-27" />

<title>BubbleTree Tutorial</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; }
code > span.dt { color: #902000; }
code > span.dv { color: #40a070; }
code > span.bn { color: #40a070; }
code > span.fl { color: #40a070; }
code > span.ch { color: #4070a0; }
code > span.st { color: #4070a0; }
code > span.co { color: #60a0b0; font-style: italic; }
code > span.ot { color: #007020; }
code > span.al { color: #ff0000; font-weight: bold; }
code > span.fu { color: #06287e; }
code > span.er { color: #ff0000; font-weight: bold; }
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<link href="data:text/css,body%20%7B%0A%20%20background%2Dcolor%3A%20%23fff%3B%0A%20%20margin%3A%201em%20auto%3B%0A%20%20max%2Dwidth%3A%20700px%3B%0A%20%20overflow%3A%20visible%3B%0A%20%20padding%2Dleft%3A%202em%3B%0A%20%20padding%2Dright%3A%202em%3B%0A%20%20font%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0A%20%20font%2Dsize%3A%2014px%3B%0A%20%20line%2Dheight%3A%201%2E35%3B%0A%7D%0A%0A%23header%20%7B%0A%20%20text%2Dalign%3A%20center%3B%0A%7D%0A%0A%23TOC%20%7B%0A%20%20clear%3A%20both%3B%0A%20%20margin%3A%200%200%2010px%2010px%3B%0A%20%20padding%3A%204px%3B%0A%20%20width%3A%20400px%3B%0A%20%20border%3A%201px%20solid%20%23CCCCCC%3B%0A%20%20border%2Dradius%3A%205px%3B%0A%0A%20%20background%2Dcolor%3A%20%23f6f6f6%3B%0A%20%20font%2Dsize%3A%2013px%3B%0A%20%20line%2Dheight%3A%201%2E3%3B%0A%7D%0A%20%20%23TOC%20%2Etoctitle%20%7B%0A%20%20%20%20font%2Dweight%3A%20bold%3B%0A%20%20%20%20font%2Dsize%3A%2015px%3B%0A%20%20%20%20margin%2Dleft%3A%205px%3B%0A%20%20%7D%0A%0A%20%20%23TOC%20ul%20%7B%0A%20%20%20%20padding%2Dleft%3A%2040px%3B%0A%20%20%20%20margin%2Dleft%3A%20%2D1%2E5em%3B%0A%20%20%20%20margin%2Dtop%3A%205px%3B%0A%20%20%20%20margin%2Dbottom%3A%205px%3B%0A%20%20%7D%0A%20%20%23TOC%20ul%20ul%20%7B%0A%20%20%20%20margin%2Dleft%3A%20%2D2em%3B%0A%20%20%7D%0A%20%20%23TOC%20li%20%7B%0A%20%20%20%20line%2Dheight%3A%2016px%3B%0A%20%20%7D%0A%0Atable%20%7B%0A%20%20margin%3A%201em%20auto%3B%0A%20%20border%2Dwidth%3A%201px%3B%0A%20%20border%2Dcolor%3A%20%23DDDDDD%3B%0A%20%20border%2Dstyle%3A%20outset%3B%0A%20%20border%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0A%20%20border%2Dwidth%3A%202px%3B%0A%20%20padding%3A%205px%3B%0A%20%20border%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0A%20%20border%2Dwidth%3A%201px%3B%0A%20%20border%2Dstyle%3A%20inset%3B%0A%20%20line%2Dheight%3A%2018px%3B%0A%20%20padding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0A%20%20border%2Dleft%2Dstyle%3A%20none%3B%0A%20%20border%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0A%20%20background%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0A%0Ap%20%7B%0A%20%20margin%3A%200%2E5em%200%3B%0A%7D%0A%0Ablockquote%20%7B%0A%20%20background%2Dcolor%3A%20%23f6f6f6%3B%0A%20%20padding%3A%200%2E25em%200%2E75em%3B%0A%7D%0A%0Ahr%20%7B%0A%20%20border%2Dstyle%3A%20solid%3B%0A%20%20border%3A%20none%3B%0A%20%20border%2Dtop%3A%201px%20solid%20%23777%3B%0A%20%20margin%3A%2028px%200%3B%0A%7D%0A%0Adl%20%7B%0A%20%20margin%2Dleft%3A%200%3B%0A%7D%0A%20%20dl%20dd%20%7B%0A%20%20%20%20margin%2Dbottom%3A%2013px%3B%0A%20%20%20%20margin%2Dleft%3A%2013px%3B%0A%20%20%7D%0A%20%20dl%20dt%20%7B%0A%20%20%20%20font%2Dweight%3A%20bold%3B%0A%20%20%7D%0A%0Aul%20%7B%0A%20%20margin%2Dtop%3A%200%3B%0A%7D%0A%20%20ul%20li%20%7B%0A%20%20%20%20list%2Dstyle%3A%20circle%20outside%3B%0A%20%20%7D%0A%20%20ul%20ul%20%7B%0A%20%20%20%20margin%2Dbottom%3A%200%3B%0A%20%20%7D%0A%0Apre%2C%20code%20%7B%0A%20%20background%2Dcolor%3A%20%23f7f7f7%3B%0A%20%20border%2Dradius%3A%203px%3B%0A%20%20color%3A%20%23333%3B%0A%7D%0Apre%20%7B%0A%20%20white%2Dspace%3A%20pre%2Dwrap%3B%20%20%20%20%2F%2A%20Wrap%20long%20lines%20%2A%2F%0A%20%20border%2Dradius%3A%203px%3B%0A%20%20margin%3A%205px%200px%2010px%200px%3B%0A%20%20padding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0A%20%20background%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0A%0Acode%20%7B%0A%20%20font%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0A%20%20font%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0A%20%20padding%3A%202px%200px%3B%0A%7D%0A%0Adiv%2Efigure%20%7B%0A%20%20text%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0A%20%20background%2Dcolor%3A%20%23FFFFFF%3B%0A%20%20padding%3A%202px%3B%0A%20%20border%3A%201px%20solid%20%23DDDDDD%3B%0A%20%20border%2Dradius%3A%203px%3B%0A%20%20border%3A%201px%20solid%20%23CCCCCC%3B%0A%20%20margin%3A%200%205px%3B%0A%7D%0A%0Ah1%20%7B%0A%20%20margin%2Dtop%3A%200%3B%0A%20%20font%2Dsize%3A%2035px%3B%0A%20%20line%2Dheight%3A%2040px%3B%0A%7D%0A%0Ah2%20%7B%0A%20%20border%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0A%20%20padding%2Dtop%3A%2010px%3B%0A%20%20padding%2Dbottom%3A%202px%3B%0A%20%20font%2Dsize%3A%20145%25%3B%0A%7D%0A%0Ah3%20%7B%0A%20%20border%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0A%20%20padding%2Dtop%3A%2010px%3B%0A%20%20font%2Dsize%3A%20120%25%3B%0A%7D%0A%0Ah4%20%7B%0A%20%20border%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0A%20%20margin%2Dleft%3A%208px%3B%0A%20%20font%2Dsize%3A%20105%25%3B%0A%7D%0A%0Ah5%2C%20h6%20%7B%0A%20%20border%2Dbottom%3A%201px%20solid%20%23ccc%3B%0A%20%20font%2Dsize%3A%20105%25%3B%0A%7D%0A%0Aa%20%7B%0A%20%20color%3A%20%230033dd%3B%0A%20%20text%2Ddecoration%3A%20none%3B%0A%7D%0A%20%20a%3Ahover%20%7B%0A%20%20%20%20color%3A%20%236666ff%3B%20%7D%0A%20%20a%3Avisited%20%7B%0A%20%20%20%20color%3A%20%23800080%3B%20%7D%0A%20%20a%3Avisited%3Ahover%20%7B%0A%20%20%20%20color%3A%20%23BB00BB%3B%20%7D%0A%20%20a%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0A%20%20%20%20text%2Ddecoration%3A%20underline%3B%20%7D%0A%20%20a%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0A%20%20%20%20text%2Ddecoration%3A%20underline%3B%20%7D%0A%0A%2F%2A%20Class%20described%20in%20https%3A%2F%2Fbenjeffrey%2Ecom%2Fposts%2Fpandoc%2Dsyntax%2Dhighlighting%2Dcss%0A%20%20%20Colours%20from%20https%3A%2F%2Fgist%2Egithub%2Ecom%2Frobsimmons%2F1172277%20%2A%2F%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%2F%2A%20Keyword%20%2A%2F%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%2F%2A%20DataType%20%2A%2F%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%2F%2A%20DecVal%20%28decimal%20values%29%20%2A%2F%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20BaseN%20%2A%2F%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20Float%20%2A%2F%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20Char%20%2A%2F%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20String%20%2A%2F%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%2F%2A%20Comment%20%2A%2F%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%2F%2A%20OtherToken%20%2A%2F%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%2F%2A%20AlertToken%20%2A%2F%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%2F%2A%20Function%20calls%20%2A%2F%20%0Acode%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%2F%2A%20ErrorTok%20%2A%2F%0A%0A" rel="stylesheet" type="text/css" />

</head>

<body>



<div id="header">
<h1 class="title">BubbleTree Tutorial</h1>
<h4 class="author"><em>Wei Zhu, Michael Kuziora, Todd Creasy, Brandon Higgs</em></h4>
<h4 class="date"><em>2015-08-27</em></h4>
</div>


<div id="abstract" class="section level2">
<h2>Abstract</h2>
<p>To evaluate the aneuploidy and prevalence of clonal or quasiclonal tumors, we developed a novel tool to characterize the mosaic tumor genome on the basis of one major assumption: the genome of a heterogeneous multi-cell tumor biopsy can be sliced into a chain of segments that are characterized by homogeneous somatic copy number alternations (SCNAs) and B allele frequencies (BAFs). The model, termed BubbleTree, utilizes both SCNAs and the interconnected BAFs as markers of tumor clones to extract tumor clonality estimates. BubbleTree is an intuitive and powerful approach to jointly identify ASCN, tumor purity and (sub)clonality, which aims to improve upon current methods to characterize the tumor karyotypes and ultimately better inform cancer diagnosis, prognosis and treatment decisions.</p>
</div>
<div id="quickstart-to-using-bubbletree" class="section level2">
<h2>Quickstart to Using BubbleTree</h2>
<p>To perform a BubbleTree analysis, data pertaining to the position and B allele frequency of heterozygous snps in the tumor sample and segmented copy number information including the position, number of markers/segment and log2 copy number ratio between tumor and normal samples must first be obtained. Example data in the desired format is provided as part of this package as GRanges objects and can be loaded as follows.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(BubbleTree)</code></pre>
<pre><code>## Loading required package: IRanges
## Loading required package: BiocGenerics
## Loading required package: parallel
## 
## Attaching package: 'BiocGenerics'
## 
## The following objects are masked from 'package:parallel':
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB
## 
## The following objects are masked from 'package:stats':
## 
##     IQR, mad, xtabs
## 
## The following objects are masked from 'package:base':
## 
##     Filter, Find, Map, Position, Reduce, anyDuplicated, append,
##     as.data.frame, as.vector, cbind, colnames, do.call,
##     duplicated, eval, evalq, get, grep, grepl, intersect,
##     is.unsorted, lapply, mapply, match, mget, order, paste, pmax,
##     pmax.int, pmin, pmin.int, rank, rbind, rep.int, rownames,
##     sapply, setdiff, sort, table, tapply, union, unique, unlist,
##     unsplit
## 
## Loading required package: S4Vectors
## Loading required package: stats4
## Loading required package: GenomicRanges
## Loading required package: GenomeInfoDb
## Loading required package: plyr
## 
## Attaching package: 'plyr'
## 
## The following object is masked from 'package:IRanges':
## 
##     desc
## 
## The following object is masked from 'package:S4Vectors':
## 
##     rename
## 
## Loading required package: dplyr
## 
## Attaching package: 'dplyr'
## 
## The following objects are masked from 'package:plyr':
## 
##     arrange, count, desc, failwith, id, mutate, rename, summarise,
##     summarize
## 
## The following objects are masked from 'package:GenomicRanges':
## 
##     intersect, setdiff, union
## 
## The following object is masked from 'package:GenomeInfoDb':
## 
##     intersect
## 
## The following objects are masked from 'package:IRanges':
## 
##     collapse, desc, intersect, setdiff, slice, union
## 
## The following objects are masked from 'package:S4Vectors':
## 
##     intersect, rename, setdiff, union
## 
## The following objects are masked from 'package:BiocGenerics':
## 
##     combine, intersect, setdiff, union
## 
## The following objects are masked from 'package:stats':
## 
##     filter, lag
## 
## The following objects are masked from 'package:base':
## 
##     intersect, setdiff, setequal, union
## 
## Loading required package: magrittr</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="kw">system.file</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;allCall.lst.RData&quot;</span>, <span class="dt">package=</span><span class="st">&quot;BubbleTree&quot;</span>))

btreeplotter &lt;-<span class="st"> </span><span class="kw">new</span>(<span class="st">&quot;BTreePlotter&quot;</span>, <span class="dt">max.ploidy=</span><span class="dv">5</span>, <span class="dt">max.size=</span><span class="dv">10</span>)

pdf.filename &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;adjplot_all&quot;</span>, <span class="st">&quot;pdf&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;.&quot;</span>)
<span class="kw">pdf</span>(pdf.filename, <span class="dt">width=</span><span class="dv">6</span>, <span class="dt">height=</span><span class="dv">4</span>)

<span class="kw">l_ply</span>(<span class="kw">names</span>(allCall.lst), function(nn){
    <span class="kw">cat</span>(<span class="st">&quot;Processing sample name: &quot;</span>, nn, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)
    rbd1 &lt;-<span class="st"> </span>allCall.lst[[nn]]@rbd
    rbd2 &lt;-<span class="st"> </span>allCall.lst[[nn]]@rbd.adj
    arrows &lt;-<span class="st"> </span><span class="kw">trackBTree</span>(btreeplotter, rbd1, rbd2, <span class="dt">min.srcSize=</span><span class="fl">0.01</span>, <span class="dt">min.trtSize=</span><span class="fl">0.01</span>)
    btree &lt;-<span class="st"> </span><span class="kw">drawBTree</span>(btreeplotter, rbd1) +<span class="st"> </span><span class="kw">drawBubbles</span>(btreeplotter, rbd2, <span class="st">&quot;gray80&quot;</span>) +<span class="st"> </span>arrows
    <span class="kw">print</span>(btree)
})</code></pre>
<pre><code>## Processing sample name:  sam2</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 1.9, 1.7</code></pre>
<pre><code>## Warning in drawBubbles(btreeplotter, rbd2, &quot;gray80&quot;): More ploidy might be suggested: 1.7, 1.5</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  sam3</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 1.6</code></pre>
<pre><code>## Warning in drawBubbles(btreeplotter, rbd2, &quot;gray80&quot;): More ploidy might be suggested: 1.5</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  sam4</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  sam5</code></pre>
<pre><code>## Processing sample name:  sam6</code></pre>
<pre><code>## Processing sample name:  sam7</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 1.7, 1.5</code></pre>
<pre><code>## Warning in drawBubbles(btreeplotter, rbd2, &quot;gray80&quot;): More ploidy might be suggested: 1.6</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  sam8</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 1.7, 1.5</code></pre>
<pre><code>## Warning in drawBubbles(btreeplotter, rbd2, &quot;gray80&quot;): More ploidy might be suggested: 1.6</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  sam9</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  sam10</code></pre>
<pre><code>## Warning: Removed 5 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  sam11</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 1.7, 1.5</code></pre>
<pre><code>## Warning in drawBubbles(btreeplotter, rbd2, &quot;gray80&quot;): More ploidy might be suggested: 1.6</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  sam12</code></pre>
<pre><code>## Warning: Removed 11 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 9 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  sam13</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 1.7, 1.5</code></pre>
<pre><code>## Warning in drawBubbles(btreeplotter, rbd2, &quot;gray80&quot;): More ploidy might be suggested: 1.6</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  sam14</code></pre>
<pre><code>## Processing sample name:  sam15</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0137-01A-01D-1490-08</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0145-01A-01D-1490-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 29, 3.3</code></pre>
<pre><code>## Warning in drawBubbles(btreeplotter, rbd2, &quot;gray80&quot;): More ploidy might be suggested: 30.2, 3.4</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0145-01A-01W-0224-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 3.3</code></pre>
<pre><code>## Warning in drawBubbles(btreeplotter, rbd2, &quot;gray80&quot;): More ploidy might be suggested: 3.4</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 5 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0171-01A-02D-1491-08</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0174-01A-01D-1491-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 22.6, 13.9, 11</code></pre>
<pre><code>## Warning in drawBubbles(btreeplotter, rbd2, &quot;gray80&quot;): More ploidy might be suggested: 22.3, 13.7, 10.9</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0185-01A-01D-1491-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 9.6</code></pre>
<pre><code>## Warning in drawBubbles(btreeplotter, rbd2, &quot;gray80&quot;): More ploidy might be suggested: 9.5</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 5 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0188-01A-01D-1491-08</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0188-01A-01W-0254-08</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 8 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 6 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 5 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0214-01A-02D-1491-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 3.6, 3.8</code></pre>
<pre><code>## Warning in drawBubbles(btreeplotter, rbd2, &quot;gray80&quot;): More ploidy might be suggested: 3.5, 3.8</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0219-01A-01D-1491-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 6.9, 6.8</code></pre>
<pre><code>## Warning in drawBubbles(btreeplotter, rbd2, &quot;gray80&quot;): More ploidy might be suggested: 6.8, 6.8</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-2557-01A-01D-1494-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 1.6, 4.9, 3.3, 3.2</code></pre>
<pre><code>## Warning in drawBubbles(btreeplotter, rbd2, &quot;gray80&quot;): More ploidy might be suggested: 1.7, 5, 3.4, 3.3</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-13-1500-01A-01D-0472-01</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 1.6, 2, 1.8</code></pre>
<pre><code>## Warning in drawBubbles(btreeplotter, rbd2, &quot;gray80&quot;): More ploidy might be suggested: 1.5, 1.9, 1.7</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 6 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 6 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 9 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-29-1768-01A-01W-0633-09</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 5 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-56-1622-01A-01D-1521-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 1.7, 1.9, 1.5, 2, 3.9, 2.1, 1.6</code></pre>
<pre><code>## Warning in drawBubbles(btreeplotter, rbd2, &quot;gray80&quot;): More ploidy might be suggested: 2.6, 2.8, 2.2, 2.9, 2.1, 1.6, 5.8, 3.1, 2.3, 1.8, 1.8, 1.6, 1.8</code></pre>
<pre><code>## Warning: Removed 7 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 13 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-56-1622-01A-01W-0782-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 1.7, 1.9, 1.5, 2, 3.9, 2.1, 1.6</code></pre>
<pre><code>## Warning in drawBubbles(btreeplotter, rbd2, &quot;gray80&quot;): More ploidy might be suggested: 2.6, 2.9, 2.2, 3, 2.1, 1.6, 5.9, 3.1, 2.3, 1.9, 1.9, 1.6, 1.8</code></pre>
<pre><code>## Warning: Removed 7 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 13 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 5 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-A2-A0EU-01A-22W-A071-09</code></pre>
<pre><code>## Warning: Removed 6 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 10 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 5 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 6 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 6 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 8 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-AO-A0JF-01A-11W-A071-09</code></pre>
<pre><code>## Warning: Removed 8 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 8 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 8 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 6 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-AO-A0JJ-01A-11W-A071-09</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 1.5</code></pre>
<pre><code>## Warning in drawBubbles(btreeplotter, rbd2, &quot;gray80&quot;): More ploidy might be suggested: 1.5, 1.6</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 9 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 14 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 11 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 7 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 6 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 5 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 6 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 10 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 8 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 5 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 5 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-BH-A0W5-01A-11D-A10G-09</code></pre>
<pre><code>## Processing sample name:  lung.wgs</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 7 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 9 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  ovary.wgs</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  lung.wes</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  ovary.wes</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  HCC4.Primary.Tumor</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  HCC4.Recurrent.Tumor</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  HCC11.Primary.Tumor</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  HCC11.Recurrent.Tumor</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dev.off</span>()</code></pre>
<pre><code>## png 
##   2</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">pdf.filename &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;adjplot_single&quot;</span>, <span class="st">&quot;pdf&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;.&quot;</span>)
<span class="kw">pdf</span>(pdf.filename, <span class="dt">width=</span><span class="dv">8</span>, <span class="dt">height=</span><span class="dv">6</span>)

<span class="kw">l_ply</span>(<span class="kw">names</span>(allCall.lst), function(nn){
    <span class="kw">cat</span>(<span class="st">&quot;Processing sample name: &quot;</span>, nn, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)
    rbd1 &lt;-<span class="st"> </span>allCall.lst[[nn]]@rbd
    rbd2 &lt;-<span class="st"> </span>allCall.lst[[nn]]@rbd.adj
    arrows &lt;-<span class="st"> </span><span class="kw">trackBTree</span>(btreeplotter, rbd1, rbd2, <span class="dt">min.srcSize =</span> <span class="fl">0.1</span>, <span class="dt">min.trtSize=</span> <span class="fl">0.1</span>)
    btree &lt;-<span class="st"> </span><span class="kw">drawBTree</span>(btreeplotter, rbd1)  +<span class="st"> </span>arrows
    <span class="kw">print</span>(btree)
})</code></pre>
<pre><code>## Processing sample name:  sam2</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 1.9, 1.7</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  sam3</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 1.6</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  sam4</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  sam5</code></pre>
<pre><code>## Processing sample name:  sam6</code></pre>
<pre><code>## Processing sample name:  sam7</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 1.7, 1.5</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  sam8</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 1.7, 1.5</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  sam9</code></pre>
<pre><code>## Processing sample name:  sam10</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  sam11</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 1.7, 1.5</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  sam12</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  sam13</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 1.7, 1.5</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  sam14</code></pre>
<pre><code>## Processing sample name:  sam15</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0137-01A-01D-1490-08</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0145-01A-01D-1490-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 29, 3.3</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0145-01A-01W-0224-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 3.3</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0171-01A-02D-1491-08</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0174-01A-01D-1491-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 22.6, 13.9, 11</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0185-01A-01D-1491-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 9.6</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0188-01A-01D-1491-08</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0188-01A-01W-0254-08</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 5 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0214-01A-02D-1491-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 3.6, 3.8</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0219-01A-01D-1491-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 6.9, 6.8</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-2557-01A-01D-1494-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 1.6, 4.9, 3.3, 3.2</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  TCGA-13-1500-01A-01D-0472-01</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 1.6, 2, 1.8</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  TCGA-29-1768-01A-01W-0633-09</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-56-1622-01A-01D-1521-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 1.7, 1.9, 1.5, 2, 3.9, 2.1, 1.6</code></pre>
<pre><code>## Warning: Removed 7 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-56-1622-01A-01W-0782-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 1.7, 1.9, 1.5, 2, 3.9, 2.1, 1.6</code></pre>
<pre><code>## Warning: Removed 7 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-A2-A0EU-01A-22W-A071-09</code></pre>
<pre><code>## Processing sample name:  TCGA-AO-A0JF-01A-11W-A071-09</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-AO-A0JJ-01A-11W-A071-09</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, rbd1): More ploidy might be suggested: 1.5</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-BH-A0W5-01A-11D-A10G-09</code></pre>
<pre><code>## Processing sample name:  lung.wgs</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  ovary.wgs</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  lung.wes</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  ovary.wes</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  HCC4.Primary.Tumor</code></pre>
<pre><code>## Processing sample name:  HCC4.Recurrent.Tumor</code></pre>
<pre><code>## Processing sample name:  HCC11.Primary.Tumor</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  HCC11.Recurrent.Tumor</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dev.off</span>()</code></pre>
<pre><code>## png 
##   2</code></pre>
<p>To generate a Excel report of a BubbleTree analysis.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(BubbleTree)
<span class="kw">load</span>(<span class="kw">system.file</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;allRBD.lst.RData&quot;</span>, <span class="dt">package=</span><span class="st">&quot;BubbleTree&quot;</span>))
btreepredictor &lt;-<span class="st"> </span><span class="kw">new</span>(<span class="st">&quot;BTreePredictor&quot;</span>)
btreepredictor@config$cutree.h &lt;-<span class="st"> </span><span class="fl">0.15</span>

high.ploidy &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="ot">TRUE</span>, <span class="kw">length</span>(allRBD.lst))
high.purity &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="ot">TRUE</span>, <span class="kw">length</span>(allRBD.lst))

high.ploidy[<span class="kw">c</span>(<span class="st">&quot;sam6&quot;</span>,
              <span class="st">&quot;ovary.wgs&quot;</span>,
              <span class="st">&quot;ovary.wes&quot;</span>,
              <span class="st">&quot;TCGA-06-0145-01A-01W-0224-08&quot;</span>,
              <span class="st">&quot;TCGA-13-1500-01A-01D-0472-01&quot;</span>,
              <span class="st">&quot;TCGA-AO-A0JJ-01A-11W-A071-09&quot;</span>)] &lt;-<span class="st"> </span><span class="ot">FALSE</span>

high.purity[<span class="kw">c</span>(<span class="st">&quot;sam6&quot;</span>, <span class="st">&quot;ovary.wgs&quot;</span>, <span class="st">&quot;ovary.wes&quot;</span>)] &lt;-<span class="st"> </span><span class="ot">FALSE</span>


allCall.lst &lt;-<span class="st"> </span>plyr::<span class="kw">llply</span>(<span class="kw">names</span>(allRBD.lst), function(nn) {
    <span class="kw">cat</span>(<span class="st">&quot;Processing sample name: &quot;</span>, nn, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)
    rbd &lt;-<span class="st"> </span>allRBD.lst[[nn]]
    btreepredictor@config$high.ploidy &lt;-<span class="st"> </span>high.ploidy[nn]
    btreepredictor@config$high.purity &lt;-<span class="st"> </span>high.purity[nn]
    btreepredictor &lt;-<span class="st"> </span><span class="kw">loadRBD</span>(btreepredictor, rbd)
    btreepredictor@config$min.segSize &lt;-<span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">max</span>(btreepredictor@rbd$seg.size, <span class="dt">na.rm=</span><span class="ot">TRUE</span>) &lt;<span class="st"> </span><span class="fl">0.4</span>, <span class="fl">0.1</span>, <span class="fl">0.4</span>)
    
    btreepredictor &lt;-<span class="st"> </span><span class="kw">btpredict</span>(btreepredictor)
    
    <span class="kw">cat</span>(<span class="kw">info</span>(btreepredictor), <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)
    <span class="kw">return</span>(btreepredictor)
})</code></pre>
<pre><code>## Processing sample name:  sam2 
## Purity: 1; Ploidy: 1.8; Deviation: 0.01 
## Processing sample name:  sam3 
## Purity: 0.82, 0.2; Ploidy: 1.8; Deviation: 0.01 
## Processing sample name:  sam4 
## Purity: 0.6; Ploidy: 1.8; Deviation: 0.01 
## Processing sample name:  sam5 
## Purity: 0.41; Ploidy: 1.8; Deviation: 0.01 
## Processing sample name:  sam6 
## Purity: 0.25; Ploidy: 1.9; Deviation: 0.02 
## Processing sample name:  sam7 
## Purity: 0.9, 0.45; Ploidy: 1.8; Deviation: 0.01 
## Processing sample name:  sam8 
## Purity: 0.9, 0.44; Ploidy: 1.8; Deviation: 0.01 
## Processing sample name:  sam9 
## Purity: 0.45; Ploidy: 1.8; Deviation: 0.02 
## Processing sample name:  sam10 
## Purity: 0.62; Ploidy: 1.8; Deviation: 0.02 
## Processing sample name:  sam11 
## Purity: 0.89, 0.6, 0.3; Ploidy: 1.8; Deviation: 0.01 
## Processing sample name:  sam12 
## Purity: 0.67, 0.31; Ploidy: 1.8; Deviation: 0.02 
## Processing sample name:  sam13 
## Purity: 0.9, 0.45; Ploidy: 1.8; Deviation: 0.01 
## Processing sample name:  sam14 
## Purity: 0.73; Ploidy: 3.0; Deviation: 0.00 
## Processing sample name:  sam15 
## Purity: 0.7; Ploidy: 4.0; Deviation: 0.02 
## Processing sample name:  TCGA-06-0137-01A-01D-1490-08 
## Purity: 0.85; Ploidy: 2.0; Deviation: 0.02 
## Processing sample name:  TCGA-06-0145-01A-01D-1490-08 
## Purity: 0.78, 0.2; Ploidy: 2.1; Deviation: 0.02 
## Processing sample name:  TCGA-06-0145-01A-01W-0224-08 
## Purity: 0.73, 0.4, 0.22; Ploidy: 2.1; Deviation: 0.02 
## Processing sample name:  TCGA-06-0171-01A-02D-1491-08 
## Purity: 0.63, 0.45, 0.21; Ploidy: 1.9; Deviation: 0.01 
## Processing sample name:  TCGA-06-0174-01A-01D-1491-08 
## Purity: 0.84, 0.49, 0.23; Ploidy: 2.0; Deviation: 0.01 
## Processing sample name:  TCGA-06-0185-01A-01D-1491-08 
## Purity: 0.84, 0.37, 0.2; Ploidy: 2.0; Deviation: 0.02 
## Processing sample name:  TCGA-06-0188-01A-01D-1491-08 
## Purity: 0.8, 0.61, 0.37, 0.23; Ploidy: 2.0; Deviation: 0.02 
## Processing sample name:  TCGA-06-0188-01A-01W-0254-08 
## Purity: 0.76, 0.65, 0.51, 0.39, 0.23; Ploidy: 2.0; Deviation: 0.03 
## Processing sample name:  TCGA-06-0214-01A-02D-1491-08 
## Purity: 0.67, 0.49, 0.22; Ploidy: 2.0; Deviation: 0.02 
## Processing sample name:  TCGA-06-0219-01A-01D-1491-08 
## Purity: 0.65, 0.21; Ploidy: 2.0; Deviation: 0.02 
## Processing sample name:  TCGA-06-2557-01A-01D-1494-08 
## Purity: 0.7, 0.22; Ploidy: 2.0; Deviation: 0.02 
## Processing sample name:  TCGA-13-1500-01A-01D-0472-01 
## Purity: 0.77, 0.4, 0.2; Ploidy: 1.8; Deviation: 0.01 
## Processing sample name:  TCGA-29-1768-01A-01W-0633-09 
## Purity: 0.56, 0.49, 0.37, 0.22; Ploidy: 1.9; Deviation: 0.01 
## Processing sample name:  TCGA-56-1622-01A-01D-1521-08 
## Purity: 0.88, 0.72, 0.4; Ploidy: 3.1; Deviation: 0.02 
## Processing sample name:  TCGA-56-1622-01A-01W-0782-08 
## Purity: 0.89, 0.73, 0.66, 0.34; Ploidy: 3.1; Deviation: 0.02 
## Processing sample name:  TCGA-A2-A0EU-01A-22W-A071-09 
## Purity: 0.8, 0.61, 0.48, 0.36; Ploidy: 1.7; Deviation: 0.01 
## Processing sample name:  TCGA-AO-A0JF-01A-11W-A071-09 
## Purity: 0.59, 0.39; Ploidy: 2.1; Deviation: 0.00 
## Processing sample name:  TCGA-AO-A0JJ-01A-11W-A071-09 
## Purity: 0.51; Ploidy: 2.1; Deviation: 0.00 
## Processing sample name:  TCGA-BH-A0W5-01A-11D-A10G-09 
## Purity: 0.71, 0.26; Ploidy: 2.0; Deviation: 0.02 
## Processing sample name:  lung.wgs 
## Purity: 0.74, 0.38, 0.2; Ploidy: 1.9; Deviation: 0.00 
## Processing sample name:  ovary.wgs 
## Purity: 0.3; Ploidy: 1.8; Deviation: 0.00 
## Processing sample name:  lung.wes 
## Purity: 0.74, 0.36, 0.2; Ploidy: 1.9; Deviation: 0.02 
## Processing sample name:  ovary.wes 
## Purity: 0.2; Ploidy: 2.0; Deviation: 0.03 
## Processing sample name:  HCC4.Primary.Tumor 
## Purity: 0.76; Ploidy: 2.0; Deviation: 0.02 
## Processing sample name:  HCC4.Recurrent.Tumor 
## Purity: 0.86, 0.2; Ploidy: 2.0; Deviation: 0.02 
## Processing sample name:  HCC11.Primary.Tumor 
## Purity: 0.8, 0.43, 0.34, 0.2; Ploidy: 1.9; Deviation: 0.01 
## Processing sample name:  HCC11.Recurrent.Tumor 
## Purity: 0.9, 0.43, 0.35, 0.2; Ploidy: 2.0; Deviation: 0.02</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">names</span>(allCall.lst) &lt;-<span class="st"> </span><span class="kw">names</span>(allRBD.lst)
results &lt;-<span class="st"> </span><span class="kw">list</span>()
for (name in <span class="kw">names</span>(allCall.lst)) {
    results[[name]] &lt;-<span class="st"> </span>allCall.lst[[name]]@result$dist
}

xls.filename &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;all_calls_report&quot;</span>, <span class="st">&quot;xlsx&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;.&quot;</span>)
<span class="kw">print</span>(xls.filename)</code></pre>
<pre><code>## [1] &quot;all_calls_report.xlsx&quot;</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">saveXLS</span>(results, xls.filename)</code></pre>
<pre><code>## Warning in saveXLS(results, xls.filename): Be cautious of embedded quotes (using remove.embedded.quotes)!!!</code></pre>
<p>To perform a BubbleTree analysis with an overlay of 77 common cancer genes.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(BubbleTree)
<span class="kw">load</span>(<span class="kw">system.file</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;allCall.lst.RData&quot;</span>, <span class="dt">package=</span><span class="st">&quot;BubbleTree&quot;</span>))
<span class="kw">load</span>(<span class="kw">system.file</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;cancer.genes.minus2.rda&quot;</span>, <span class="dt">package=</span><span class="st">&quot;BubbleTree&quot;</span>))
<span class="kw">load</span>(<span class="kw">system.file</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;vol.genes.rda&quot;</span>, <span class="dt">package=</span><span class="st">&quot;BubbleTree&quot;</span>))
<span class="kw">load</span>(<span class="kw">system.file</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;gene.uni.clean.gr.rda&quot;</span>, <span class="dt">package=</span><span class="st">&quot;BubbleTree&quot;</span>))
<span class="kw">load</span>(<span class="kw">system.file</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;cyto.gr.rda&quot;</span>, <span class="dt">package=</span><span class="st">&quot;BubbleTree&quot;</span>))

<span class="co"># 77 common cancer genes</span>
comm &lt;-<span class="st"> </span><span class="kw">btcompare</span>(vol.genes, cancer.genes.minus2)</code></pre>
<pre><code>## 77 are common, whereas 35 and 302 are unique in each of dataset</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">btreeplotter &lt;-<span class="st"> </span><span class="kw">new</span>(<span class="st">&quot;BTreePlotter&quot;</span>, <span class="dt">branch.col=</span><span class="st">&quot;gray50&quot;</span>)
annotator &lt;-<span class="kw">new</span>(<span class="st">&quot;Annotate&quot;</span>)

pdf.filename &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;bubbletree_preview&quot;</span>, <span class="st">&quot;pdf&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;.&quot;</span>)
<span class="kw">pdf</span>(pdf.filename, <span class="dt">width=</span><span class="dv">10</span>, <span class="dt">height=</span><span class="dv">6</span>)

allPreview &lt;-<span class="st"> </span><span class="kw">llply</span>(<span class="kw">names</span>(allCall.lst), function(nn){
    <span class="kw">cat</span>(<span class="st">&quot;Processing sample name: &quot;</span>, nn, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)
    cc &lt;-<span class="st"> </span>allCall.lst[[nn]]
    z &lt;-<span class="st"> </span><span class="kw">drawBTree</span>(btreeplotter, cc@rbd.adj) +<span class="st"> </span>ggplot2::<span class="kw">labs</span>(<span class="dt">title=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s (%s)&quot;</span>, nn, <span class="kw">info</span>(cc)))
    <span class="kw">print</span>(z)
    out &lt;-<span class="st"> </span>cc@result$dist  %&gt;%<span class="st"> </span><span class="kw">filter</span>(seg.size &gt;=<span class="st"> </span><span class="fl">0.1</span> ) %&gt;%<span class="st"> </span><span class="kw">arrange</span>(gtools::<span class="kw">mixedorder</span>(<span class="kw">as.character</span>(seqnames)), start) <span class="co"># needs to be relevel</span>
    
    ann &lt;-<span class="st"> </span><span class="kw">with</span>(out, {
        <span class="kw">annoByGenesAndCyto</span>(annotator,
                           <span class="kw">as.character</span>(out$seqnames),
                           <span class="kw">as.numeric</span>(out$start),
                           <span class="kw">as.numeric</span>(out$end),
                           comm$comm,
                           <span class="dt">gene.uni.clean.gr=</span>gene.uni.clean.gr,
                           <span class="dt">cyto.gr=</span>cyto.gr)
    })
    
    out$cyto &lt;-<span class="st"> </span>ann$cyto
    out$genes &lt;-<span class="st"> </span>ann$ann
    <span class="kw">return</span>(out)
})</code></pre>
<pre><code>## Processing sample name:  sam2</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, cc@rbd.adj): More ploidy might be suggested: 1.7</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  sam3</code></pre>
<pre><code>## Processing sample name:  sam4</code></pre>
<pre><code>## Processing sample name:  sam5</code></pre>
<pre><code>## Processing sample name:  sam6</code></pre>
<pre><code>## Processing sample name:  sam7</code></pre>
<pre><code>## Processing sample name:  sam8</code></pre>
<pre><code>## Processing sample name:  sam9</code></pre>
<pre><code>## Processing sample name:  sam10</code></pre>
<pre><code>## Processing sample name:  sam11</code></pre>
<pre><code>## Processing sample name:  sam12</code></pre>
<pre><code>## Processing sample name:  sam13</code></pre>
<pre><code>## Processing sample name:  sam14</code></pre>
<pre><code>## Processing sample name:  sam15</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0137-01A-01D-1490-08</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0145-01A-01D-1490-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, cc@rbd.adj): More ploidy might be suggested: 30.2, 3.4</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0145-01A-01W-0224-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, cc@rbd.adj): More ploidy might be suggested: 3.4</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0171-01A-02D-1491-08</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0174-01A-01D-1491-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, cc@rbd.adj): More ploidy might be suggested: 22.3, 13.7, 10.9</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0185-01A-01D-1491-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, cc@rbd.adj): More ploidy might be suggested: 9.5</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0188-01A-01D-1491-08</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0188-01A-01W-0254-08</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0214-01A-02D-1491-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, cc@rbd.adj): More ploidy might be suggested: 3.5, 3.8</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0219-01A-01D-1491-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, cc@rbd.adj): More ploidy might be suggested: 6.8, 6.8</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-2557-01A-01D-1494-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, cc@rbd.adj): More ploidy might be suggested: 5, 3.4, 3.3</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  TCGA-13-1500-01A-01D-0472-01</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, cc@rbd.adj): More ploidy might be suggested: 1.9, 1.7</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  TCGA-29-1768-01A-01W-0633-09</code></pre>
<pre><code>## Processing sample name:  TCGA-56-1622-01A-01D-1521-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, cc@rbd.adj): More ploidy might be suggested: 2.6, 2.8, 2.2, 2.9, 2.1, 5.8, 3.1, 2.3, 1.8, 1.8, 1.8</code></pre>
<pre><code>## Warning: Removed 11 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  TCGA-56-1622-01A-01W-0782-08</code></pre>
<pre><code>## Warning in drawBTree(btreeplotter, cc@rbd.adj): More ploidy might be suggested: 2.6, 2.9, 2.2, 3, 2.1, 5.9, 3.1, 2.3, 1.9, 1.9, 1.8</code></pre>
<pre><code>## Warning: Removed 11 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  TCGA-A2-A0EU-01A-22W-A071-09</code></pre>
<pre><code>## Processing sample name:  TCGA-AO-A0JF-01A-11W-A071-09</code></pre>
<pre><code>## Processing sample name:  TCGA-AO-A0JJ-01A-11W-A071-09</code></pre>
<pre><code>## Processing sample name:  TCGA-BH-A0W5-01A-11D-A10G-09</code></pre>
<pre><code>## Processing sample name:  lung.wgs</code></pre>
<pre><code>## Processing sample name:  ovary.wgs</code></pre>
<pre><code>## Processing sample name:  lung.wes</code></pre>
<pre><code>## Processing sample name:  ovary.wes</code></pre>
<pre><code>## Processing sample name:  HCC4.Primary.Tumor</code></pre>
<pre><code>## Processing sample name:  HCC4.Recurrent.Tumor</code></pre>
<pre><code>## Processing sample name:  HCC11.Primary.Tumor</code></pre>
<pre><code>## Processing sample name:  HCC11.Recurrent.Tumor</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dev.off</span>()</code></pre>
<pre><code>## png 
##   2</code></pre>
<p>To perform a BubbleTree analysis with an overlays of other types of genomic annotations.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(BubbleTree)
<span class="kw">load</span>(<span class="kw">system.file</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;allCall.lst.RData&quot;</span>, <span class="dt">package=</span><span class="st">&quot;BubbleTree&quot;</span>))
<span class="kw">load</span>(<span class="kw">system.file</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;centromere.dat.rda&quot;</span>, <span class="dt">package=</span><span class="st">&quot;BubbleTree&quot;</span>))
<span class="kw">load</span>(<span class="kw">system.file</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;all.somatic.lst.RData&quot;</span>, <span class="dt">package=</span><span class="st">&quot;BubbleTree&quot;</span>))
<span class="kw">load</span>(<span class="kw">system.file</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;allHetero.lst.RData&quot;</span>, <span class="dt">package=</span><span class="st">&quot;BubbleTree&quot;</span>))
<span class="kw">load</span>(<span class="kw">system.file</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;allCNV.lst.RData&quot;</span>, <span class="dt">package=</span><span class="st">&quot;BubbleTree&quot;</span>))
<span class="kw">load</span>(<span class="kw">system.file</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;hg19.seqinfo.rda&quot;</span>, <span class="dt">package=</span><span class="st">&quot;BubbleTree&quot;</span>))

trackplotter &lt;-<span class="st"> </span><span class="kw">new</span>(<span class="st">&quot;TrackPlotter&quot;</span>)

gr2 =<span class="st"> </span>centromere.dat

pdf.filename &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;all_tracks3&quot;</span>, <span class="st">&quot;pdf&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;.&quot;</span>)
<span class="kw">pdf</span>(pdf.filename, <span class="dt">width=</span><span class="dv">10</span>, <span class="dt">height=</span><span class="dv">7</span>)

plyr::<span class="kw">l_ply</span>(<span class="kw">names</span>(allCall.lst), function(nn) {
    <span class="kw">cat</span>(<span class="st">&quot;Processing sample name: &quot;</span>, nn, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)
    ymax &lt;-<span class="st"> </span><span class="kw">ifelse</span>(nn %in%<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;lung.wgs&quot;</span>, <span class="st">&quot;lung.wes&quot;</span>), <span class="dv">9</span>, <span class="fl">4.3</span>)
    
    p1 &lt;-<span class="st"> </span><span class="kw">xyTrack</span>(trackplotter,
                  <span class="dt">result.dat=</span>allCall.lst[[nn]]@result,
                  <span class="dt">gr2=</span>gr2,
                  <span class="dt">ymax=</span>ymax) +<span class="st"> </span>ggplot2::<span class="kw">labs</span>(<span class="dt">title=</span>nn)
    
    p2 &lt;-<span class="st"> </span><span class="kw">bafTrack</span>(trackplotter,
                   <span class="dt">result.dat=</span>allCall.lst[[nn]]@result,
                   <span class="dt">gr2=</span>gr2,
                   <span class="dt">somatic.gr=</span>all.somatic.lst[[nn]])
    
    t1 &lt;-<span class="st"> </span><span class="kw">getTracks</span>(p1, p2)
    
    z1 &lt;-<span class="st"> </span><span class="kw">heteroLociTrack</span>(trackplotter, allCall.lst[[nn]]@result, gr2, allHetero.lst[[nn]])
    z2 &lt;-<span class="st"> </span><span class="kw">RscoreTrack</span>(trackplotter, allCall.lst[[nn]]@result, gr2, allCNV.lst[[nn]])
    t2 &lt;-<span class="st"> </span><span class="kw">getTracks</span>(z1, z2)
    
    gridExtra::<span class="kw">grid.arrange</span>(t1,t2, <span class="dt">ncol=</span><span class="dv">1</span>)
    
})</code></pre>
<pre><code>## Processing sample name:  sam2</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1092 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1092 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1092 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1092 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1092 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1092 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  sam3</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  sam4</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  sam5</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  sam6</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  sam7</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  sam8</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  sam9</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  sam10</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  sam11</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 11 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  sam12</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 6 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  sam13</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  sam14</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Processing sample name:  sam15</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0137-01A-01D-1490-08</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0145-01A-01D-1490-08</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 23 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 8 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0145-01A-01W-0224-08</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 6 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 8 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0171-01A-02D-1491-08</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0174-01A-01D-1491-08</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 6 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 10 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0185-01A-01D-1491-08</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0188-01A-01D-1491-08</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0188-01A-01W-0254-08</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0214-01A-02D-1491-08</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-0219-01A-01D-1491-08</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 11 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-06-2557-01A-01D-1494-08</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 23 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 8 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-13-1500-01A-01D-0472-01</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 17 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 20 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-29-1768-01A-01W-0633-09</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 11 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 19 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-56-1622-01A-01D-1521-08</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 20 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 33 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 7 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-56-1622-01A-01W-0782-08</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 21 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 33 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 7 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-A2-A0EU-01A-22W-A071-09</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 9 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 55 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-AO-A0JF-01A-11W-A071-09</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 8 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 49 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-AO-A0JJ-01A-11W-A071-09</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 21 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 174 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  TCGA-BH-A0W5-01A-11D-A10G-09</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Processing sample name:  lung.wgs</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 9297 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  ovary.wgs</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 2091 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  lung.wes</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Processing sample name:  ovary.wes</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Processing sample name:  HCC4.Primary.Tumor</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 27 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  HCC4.Recurrent.Tumor</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 6 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 297 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  HCC11.Primary.Tumor</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 13 rows containing missing values (geom_point).</code></pre>
<pre><code>## Processing sample name:  HCC11.Recurrent.Tumor</code></pre>
<pre><code>## Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_segment).</code></pre>
<pre><code>## Warning: Removed 31 rows containing missing values (geom_point).</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dev.off</span>()</code></pre>
<pre><code>## png 
##   2</code></pre>
<p>To perform a comparison of cancer datasets.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(BubbleTree)
<span class="kw">load</span>(<span class="kw">system.file</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;allCall.lst.RData&quot;</span>, <span class="dt">package=</span><span class="st">&quot;BubbleTree&quot;</span>))

btp &lt;-<span class="st"> </span><span class="kw">new</span>(<span class="st">&quot;BTreePlotter&quot;</span>, <span class="dt">max.ploidy=</span><span class="dv">5</span>, <span class="dt">max.size=</span><span class="dv">10</span>)
pairs &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="kw">c</span>(<span class="st">&quot;lung.wes&quot;</span>, <span class="st">&quot;lung.wgs&quot;</span>),
              <span class="kw">c</span>(<span class="st">&quot;HCC4.Primary.Tumor&quot;</span>, <span class="st">&quot;HCC4.Recurrent.Tumor&quot;</span>),
              <span class="kw">c</span>(<span class="st">&quot;HCC11.Primary.Tumor&quot;</span>, <span class="st">&quot;HCC11.Recurrent.Tumor&quot;</span> ))

<span class="kw">pdf</span>(<span class="st">&quot;medi.comparePlot.single2.pdf&quot;</span>, <span class="dt">width=</span><span class="dv">8</span>,<span class="dt">height=</span><span class="dv">6</span>)

plyr::<span class="kw">l_ply</span>(pairs, function(p){
    <span class="kw">cat</span>(p, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)
    rbd1 &lt;-<span class="st"> </span>allCall.lst[[p[<span class="dv">1</span>]]]@result$dist
    rbd2 &lt;-<span class="st"> </span>allCall.lst[[p[<span class="dv">2</span>]]]@result$dist
    
    srcSize &lt;-<span class="st"> </span><span class="fl">0.5</span>
    trtSize &lt;-<span class="st"> </span><span class="kw">ifelse</span>(p[<span class="dv">1</span>] ==<span class="st"> &quot;lung.wes&quot;</span>, <span class="fl">0.1</span>, <span class="dv">1</span>)
    minOver &lt;-<span class="st"> </span><span class="kw">ifelse</span>(p[<span class="dv">1</span>] ==<span class="st"> &quot;lung.wes&quot;</span>, <span class="fl">5e6</span>, <span class="fl">1e7</span>)
    arrows &lt;-<span class="st"> </span><span class="kw">trackBTree</span>(btp,
                         rbd1,
                         rbd2,
                         <span class="dt">min.srcSize=</span>srcSize,
                         <span class="dt">min.trtSize=</span>trtSize,
                         <span class="dt">min.overlap=</span>minOver)
    
    z &lt;-<span class="st"> </span><span class="kw">drawBTree</span>(btp, rbd1)
    if(!<span class="kw">is.null</span>(arrows))
        z &lt;-<span class="st"> </span>z +<span class="st"> </span>arrows +<span class="st"> </span>ggplot2::<span class="kw">labs</span>(<span class="dt">title=</span><span class="kw">sprintf</span>(<span class="st">&quot;%s -&gt; %s&quot;</span>, p[<span class="dv">1</span>], p[<span class="dv">2</span>]))
    <span class="kw">print</span>(z)
})</code></pre>
<pre><code>## lung.wes lung.wgs</code></pre>
<pre><code>## HCC4.Primary.Tumor HCC4.Recurrent.Tumor</code></pre>
<pre><code>## HCC11.Primary.Tumor HCC11.Recurrent.Tumor</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dev.off</span>()</code></pre>
<pre><code>## png 
##   2</code></pre>
<p>To print out an Excel document of summary results.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(BubbleTree)
<span class="kw">load</span>(<span class="kw">system.file</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;allCall.lst.RData&quot;</span>, <span class="dt">package=</span><span class="st">&quot;BubbleTree&quot;</span>))

all.summary &lt;-<span class="st"> </span>plyr::<span class="kw">ldply</span>(allCall.lst, function(.Object) {
    purity &lt;-<span class="st"> </span>.Object@result$prev[<span class="dv">1</span>]
    adj &lt;-<span class="st"> </span>.Object@result$ploidy.adj[<span class="st">&quot;adj&quot;</span>]
    ploidy &lt;-<span class="st"> </span>(<span class="dv">2</span>*adj -<span class="dv">2</span>)/purity +<span class="st"> </span><span class="dv">2</span>  <span class="co"># when purity is low the calculation result is not reliable</span>
    
    <span class="kw">with</span>(.Object@result,
         <span class="kw">return</span>(<span class="kw">c</span>(<span class="dt">Purity=</span><span class="kw">round</span>(purity,<span class="dv">3</span>),
                  <span class="dt">Prevalences=</span><span class="kw">paste</span>(<span class="kw">round</span>(prev,<span class="dv">3</span>), <span class="dt">collapse=</span><span class="st">&quot;, &quot;</span>),
                  <span class="st">&quot;Tumor ploidy&quot;</span>=<span class="kw">round</span>(ploidy,<span class="dv">1</span>))))
}) %&gt;%<span class="st"> </span>plyr::<span class="kw">rename</span>(<span class="kw">c</span>(<span class="st">&quot;.id&quot;</span>=<span class="st">&quot;Sample&quot;</span>))

xls.filename &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;all_summary&quot;</span>, <span class="st">&quot;xlsx&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;.&quot;</span>)
<span class="kw">saveXLS</span>(<span class="kw">list</span>(<span class="dt">Summary=</span>all.summary), xls.filename)</code></pre>
<pre><code>## Warning in saveXLS(list(Summary = all.summary), xls.filename): Be cautious of embedded quotes (using remove.embedded.quotes)!!!</code></pre>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
